internal/services/azurestackhci/stack_hci_logical_network_resource.go

```diff
@@ -0,0 +1,485 @@
+					VMSwitchName: pointer.To(config.VirtualSwitchName),
+					Subnets:      expandStackHCILogicalNetworkSubnet(config.Subnet),
+					DhcpOptions: &logicalnetworks.LogicalNetworkPropertiesDhcpOptions{
+						DnsServers: pointer.To(config.DNSServers),
+					},
+				},
+			}
+
+			future, err := client.CreateOrUpdate(ctx, id, payload)
+			if err != nil {
+				return fmt.Errorf("performing create %s: %+v", id, err)
+			}
+
+			metadata.SetID(id)
+
+			if err := future.Poller.PollUntilDone(ctx); err != nil {
+				return fmt.Errorf("polling after create %s: %+v", id, err)
+			}
+
+			return nil
+		},
+	}
+}
+
+func (r StackHCILogicalNetworkResource) Read() sdk.ResourceFunc {
+	return sdk.ResourceFunc{
+		Timeout: 5 * time.Minute,
+		Func: func(ctx context.Context, metadata sdk.ResourceMetaData) error {
+			client := metadata.Client.AzureStackHCI.LogicalNetworks
+
+			id, err := logicalnetworks.ParseLogicalNetworkID(metadata.ResourceData.Id())
+			if err != nil {
+				return err
+			}
+
+			resp, err := client.Get(ctx, *id)
+			if err != nil {
+				if response.WasNotFound(resp.HttpResponse) {
+					return metadata.MarkAsGone(id)
+				}
+
+				return fmt.Errorf("retrieving %s: %+v", id, err)
+			}
+
+			schema := StackHCILogicalNetworkResourceModel{
+				Name:              id.LogicalNetworkName,
+				ResourceGroupName: id.ResourceGroupName,
+			}
+
+			if model := resp.Model; model != nil {
+				schema.Location = location.Normalize(model.Location)
+				schema.Tags = tags.Flatten(model.Tags)
+
+				if model.ExtendedLocation != nil && model.ExtendedLocation.Name != nil {
+					customLocationId, err := customlocations.ParseCustomLocationIDInsensitively(*model.ExtendedLocation.Name)
+					if err != nil {
+						return err
+					}
+
+					schema.CustomLocationId = customLocationId.ID()
+				}
+
+				if props := model.Properties; props != nil {
+					schema.Subnet = flattenStackHCILogicalNetworkSubnet(props.Subnets)
+					schema.VirtualSwitchName = pointer.From(props.VMSwitchName)
+
+					if props.DhcpOptions != nil {
+						schema.DNSServers = pointer.From(props.DhcpOptions.DnsServers)
+					}
+				}
+			}
+
+			return metadata.Encode(&schema)
+		},
+	}
+}
+
+func (r StackHCILogicalNetworkResource) Update() sdk.ResourceFunc {
+	return sdk.ResourceFunc{
+		Timeout: 30 * time.Minute,
+		Func: func(ctx context.Context, metadata sdk.ResourceMetaData) error {
+			client := metadata.Client.AzureStackHCI.LogicalNetworks
+
+			id, err := logicalnetworks.ParseLogicalNetworkID(metadata.ResourceData.Id())
+			if err != nil {
+				return err
+			}
+
+			var model StackHCILogicalNetworkResourceModel
+			if err := metadata.Decode(&model); err != nil {
+				return fmt.Errorf("decoding: %+v", err)
+			}
+
+			resp, err := client.Get(ctx, *id)
+			if err != nil {
+				return fmt.Errorf("retrieving %s: %+v", *id, err)
+			}
+
+			parameters := resp.Model
+			if parameters == nil {
+				return fmt.Errorf("retrieving %s: model was nil", *id)
```

Comment:
We put the object or property name between back ticks in the error message
```suggestion
				return fmt.Errorf("retrieving %s: `model` was nil", *id)
```

Reply:
added